### PR TITLE
Fix latest pylint errors

### PR DESCRIPTION
**Pull request by manoelmarques**

### CODE REVIEW

File: qiskit_nature/operators/second_quantization/vibrational_op.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def __init__(
             # Sparse label
             dense_labels = self._convert_to_dense_labels(data, num_modals)
 
-            ops: List[VibrationalOp] = []
+            ops: List["VibrationalOp"] = []
```
Comment (mrossinek) [inline qiskit_nature/operators/second_quantization/vibrational_op.py:153]: Huh.. I'm surprised to see that this supposedly worked before.. There is also the option to do
```python
from __future__ import annotations
```
What do you think about that?

**Review by mrossinek — APPROVED** [review]
LGTM! I have a minor comment about a potential alternative solution to one aspect but since it's not required I will approve this as is, already :+1:

File: qiskit_nature/operators/second_quantization/vibrational_op.py (hunk 1 above)
Comment (manoelmarques) [reply qiskit_nature/operators/second_quantization/vibrational_op.py:153]: I think the double quotes makes sense as it is what we already use everywhere in typehints with the class being declared inside itself. This fix works with both the old pylint and the new one. It is just that the old didn't report it.

Comment (manoelmarques) [reply qiskit_nature/operators/second_quantization/vibrational_op.py:153]: For the next release after Terra 0.19 we will drop support for python 3.6. The we can use `from __future__ import annotations` and remove the strings from  the type annotations

Comment (mrossinek) [reply qiskit_nature/operators/second_quantization/vibrational_op.py:153]: Right that is only available in python 3.7+.. I hadn't thought of that :sweat_smile:

**Pull request merged**